### DISCUSSION TITLE
MAP mrgsolve -  combine with model averaging algorithm (MAA)

### DISCUSSION THREAD

**Discussion by azerty89** [metrumresearchgroup/mrgsolve#890] [General]
Hello, Firstly, sorry for disturbing you.
Infact, I developed a web-based shiny tool  for  vancomycin TDM.
At this time, I'm extremely intrigued by the newly published paper on CPT: A Model Averaging/Selection Approach Enhances the Predictive Performance of Model-Informed Precision Dosing: The Case of Vancomycin - Uster - 2021 - Clinical Pharmacology & Therapeutics - Wiley Online Library.
I'm really hoping to improve my tools with these suggestions.
Due to the fact that I am not a mathematician, could you:
+ give me some existing package allowing to implement model selection algorithm (MSA) and model averaging algorithm (MAA) 
+ or  provide some instructions on how to code some things like that in combination with the MAP bayesian of mgrsolve
Thank you very much.

**Comment by FelicienLL** [comment]
Hello @azerty89, 
I allow myself to answer you since our team [developed](https://github.com/FelicienLL/mapbayr) and [validated](https://ascpt.onlinelibrary.wiley.com/doi/10.1002/psp4.12689) **mapbayr**, a package suited to performed MAP-Bayesian estimation from models defined in mrgsolve. It does not natively implement MSA/MAA algorithms, and as far as I know, there are no packages to do such thing for the specific case of PK or PK/PD models and TDM. So I guess you would have to code something by yourself. Ulster et al gave a good description of their algorithm: first perform the map-bayes estimation with every model, then compute the "weight" of each estimation from the predicted concentration, the AIC criterion or the likelihood (in mapbayr, the OFV can be found in the `$opt.value` slot of the estimate object), and finally apply these weights on your output, typically an AUC in the case of vancomycin.
I'd be happy to hear from you if you decide to employ mapbayr to do such thing, maybe directly on the issue tracker of mapbayr if you have any problem.
Have a good day
Félicien
